### PR TITLE
[yaml] Add yaml_provider.py Unit Tests

### PR DESCRIPTION
**Pull request by Polber**
Add unit tests for yaml_provider.py.

This PR also introduces non-breaking name changes to functions and variables to suppress IDE warnings.

There is also a change to the `WithSchema` built-in transform - it now requires a `schema` tag so that the schema field names do not collide with the top-level transform tags (e.g "name").

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/yaml/yaml_provider.py [hunk 1]
```diff
@@ -429,7 +441,9 @@ def parse_schema(spec):
           ],
           id=str(uuid.uuid4()))
 
-    named_tuple = schemas.named_tuple_from_schema(parse_schema(args))
+    if 'schema' not in args.keys():
+      raise ValueError("WithSchema transform missing required 'schema' tag.")
+    named_tuple = schemas.named_tuple_from_schema(parse_schema(args['schema']))
```
Comment (Polber) [inline sdks/python/apache_beam/yaml/yaml_provider.py:446]: @robertwb Should I remove this now that we have the `config` tag?

Comment (robertwb) [reply sdks/python/apache_beam/yaml/yaml_provider.py:463]: The schema needs to be in the config now.

File: sdks/python/apache_beam/yaml/yaml_provider_unit_test.py [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+    provider = ExternalProvider.provider_from_spec(spec)
+    self.java_provider_from_spec_asserts(provider)
+
+
+class ExternalPythonProviderTest(unittest.TestCase):
+  """Test class for ExternalPythonProvider tests."""
+  def __init__(self, method_name="runExternalPythonProviderTest"):
+    super().__init__(method_name)
+
+    self._type = 'pythonPackage'
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/yaml_provider_unit_test.py:488]: You could have _type as a class-level attribute and not bother overriding __init__.

**Review by robertwb — APPROVED** [review]

**Comment by AnandInguva** [comment]
@Polber is this PR still relevant?

**Pull request closed without merging**